include/ck_tile/core/utility/unary_element_function.hpp

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2023, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include "ck_tile/core.hpp"
+
+namespace ck_tile {
+
+template <typename F, typename... Fs>
+struct composer
+{
+    composer(F f, Fs... fs) : f_(f), tail_(fs...) {}
+
+    template <typename T>
+    CK_TILE_HOST_DEVICE constexpr T operator()(const T& arg) const
+    {
+        return f_(tail_(arg));
+    }
+
+    F f_;
+    composer<Fs...> tail_;
+};
+
+template <typename F>
+struct composer<F>
+{
+    composer(F f) : f_(f) {}
+
+    template <typename T>
+    CK_TILE_HOST_DEVICE constexpr T operator()(const T& arg) const
+    {
+        return f_(arg);
+    }
+
+    F f_;
+};
+
+template <typename... F>
+CK_TILE_HOST auto compose(F... f)
+{
+    return composer<F...>(f...);
+}
+
+// start of unary element function
+
+struct scale
```

Comment:
allow user pass `value_type` in future. we can simply reuse `std::multiples<>` to implement this function